packages/daemon/src/services/index.ts

```diff
@@ -461,6 +500,62 @@ export const handleVertexAccepted = async (context: Context, _event: Event) => {
       }
     }
 
+    /**
+     * Nano Contract Token Deletion Logic
+     *
+     * Handle token deletion when nano contract execution state changes.
+     *
+     * Context:
+     * - Nano contracts can create tokens via syscalls when they execute successfully
+     * - These tokens are only valid when nc_execution = 'success'
+     * - During reorgs, nano execution can be invalidated (nc_execution becomes PENDING/FAILURE/SKIPPED/null)
+     * - When this happens, any tokens created by that nano execution must be deleted
+     *
+     * Why this is needed:
+     * - A transaction with nano headers can create tokens in TWO ways:
+     *   1. Via CREATE_TOKEN_TX (token_id = tx_id) - deleted only on void
+     *   2. Via nano syscall (token_id ≠ tx_id) - deleted when nc_execution becomes non-SUCCESS
+     *
+     * Edge case: Hybrid transactions
+     * - A hybrid transaction (CREATE_TOKEN_TX + nano headers) creates BOTH types of tokens
+     * - During reorg: We must delete ONLY the nano-created tokens, NOT the CREATE_TOKEN_TX token
+     * - The getTokensCreatedByTx query returns ALL tokens created by this tx_id
+     * - However, in practice:
+     *   - Pure nano contracts: All tokens in the result are nano-created (safe to delete all)
+     *   - Hybrid transactions: Both CREATE_TOKEN_TX and nano-created tokens are in the result
+     *     BUT the CREATE_TOKEN_TX token was already deleted by a previous VERTEX_METADATA_CHANGED
+     *     event that voided the transaction, so it won't be in the database anymore
+     *
+     * Flow example (hybrid transaction during reorg):
+     * 1. CREATE_TOKEN_TX arrives → TOKEN_CREATED event (CREATE_TOKEN_TX token stored)
+     * 2. Tx gets first_block → nano executes → TOKEN_CREATED event (nano token stored)
+     * 3. VERTEX_METADATA_CHANGED → nc_execution: SUCCESS (both tokens in DB)
+     * 4. REORG → VERTEX_METADATA_CHANGED → nc_execution: PENDING, first_block: null
+     *    → This code runs → Deletes nano token, CREATE_TOKEN_TX token remains
+     * 5. Tx executes again → TOKEN_CREATED → nano token re-created (INSERT IGNORE handles duplicates)
+     *
+     * Note: This logic is INDEPENDENT of transaction voiding:
+     * - Voiding deletes ALL tokens (handled by voidTx function)
+     * - This logic deletes ONLY nano-created tokens when execution state changes
+     * - A voided transaction might still have nc_execution = 'success'
+     */
+    const hasNanoHeaders = headers && headers.length > 0 && headers.some((h) => isNanoHeader(h));
+
+    if (hasNanoHeaders) {
+      const ncExecution = metadata?.nc_execution;
+
+      // If nc_execution is not 'success', delete any tokens created by this nano contract
```

Comment:
Why? We may still execute the transaction if the state is pending or skipped, right?